### PR TITLE
soundness fixes

### PR DESCRIPTION
**Pull request by pq**
/cc @bwilkerson

### CODE REVIEW

File: lib/src/rules/prefer_single_quotes.dart [hunk 1]
```diff
@@ -141,14 +141,14 @@ class _ImmediateChildrenVisitor extends UnifyingAstVisitor {
...
-  bool? isOrContainsString(AstNode node) => node.accept(this);
+  bool isOrContainsString(AstNode node) => node.accept(this) ?? false;
 
   /// Scan as little of the tree as possible, by bailing out on first match. For
   /// all leaf nodes, they will either have a method defined here and return
   /// true, or they will return false because leaves have no children.
   @override
-  bool visitNode(AstNode node) => _ImmediateChildrenVisitor.childrenOf(node)
-      .any(isOrContainsString as bool Function(AstNode));
+  bool visitNode(AstNode node) =>
```
Comment (pq) [inline lib/src/rules/prefer_single_quotes.dart:150]: I'm pretty sure this unsafe cast was produced by the migration tool.

/fyi @srawlins

Comment (srawlins) [reply lib/src/rules/prefer_single_quotes.dart:150]: Yes, there is a known bug for these unnecessary function casts

Comment (pq) [reply lib/src/rules/prefer_single_quotes.dart:150]: Awesome.  Thanks!

**Comment by pq** [comment]
The coverage dip is really curious and feels wrong.

/fyi @kevmoo @grouma in case they've seen any oddness w/ sound-mode and coverage reporting

**Review by bwilkerson — APPROVED** [review]

**Review by parlough — APPROVED** [review]

**Pull request merged**